File: src/main/java/com/extendedae_plus/common/registry/menu/host/HostPriorityTool.java
package com.extendedae_plus.common.registry.menu.host;

import appeng.api.implementations.menuobjects.ItemMenuHost;
import appeng.menu.locator.ItemMenuHostLocator;
import com.extendedae_plus.common.init.ModDataComponents;
import com.extendedae_plus.common.registry.item.priorityTool.DataPriority;
import com.extendedae_plus.common.registry.item.priorityTool.ItemPriorityTool;
import net.minecraft.world.entity.player.Player;
import org.jetbrains.annotations.Nullable;

public class HostPriorityTool extends ItemMenuHost<ItemPriorityTool> {
    public HostPriorityTool(ItemPriorityTool item, Player player, ItemMenuHostLocator locator) {
        super(item, player, locator);

        var itemStack = this.getItemStack();
        if (!itemStack.has(ModDataComponents.DATA_PRIORITY)) {
            itemStack.set(ModDataComponents.DATA_PRIORITY,
                    new DataPriority(0, DataPriority.ModeTool.KEEP));
        }
    }

    public @Nullable DataPriority getData() {
        return this.getItemStack().get(ModDataComponents.DATA_PRIORITY);
    }

    public void setData(DataPriority data) {
        this.getItemStack().set(ModDataComponents.DATA_PRIORITY, data);
    }
}

File: src/main/kotlin/com/extendedae_plus/common/registry/menu/host/HostPriorityTool.kt
package com.extendedae_plus.common.registry.menu.host

import appeng.api.implementations.menuobjects.ItemMenuHost
import appeng.menu.locator.ItemMenuHostLocator
import com.extendedae_plus.common.init.EAEPDataComponents
import com.extendedae_plus.common.registry.item.priorityTool.DataPriority
import com.extendedae_plus.common.registry.item.priorityTool.ItemPriorityTool
import com.fish.fishlib.util.component
import com.fish.fishlib.util.extension.invoke
import net.minecraft.world.entity.player.Player

class HostPriorityTool(
    item: ItemPriorityTool, player: Player, locator: ItemMenuHostLocator
) : ItemMenuHost<ItemPriorityTool>(item, player, locator) {
    init {
        val itemStack = this.itemStack
        if (!itemStack.has(EAEPDataComponents.Priority)) {
            itemStack.set(
                EAEPDataComponents.Priority,
                DataPriority(0, DataPriority.ModeTool.Keep)
            )
        }
    }

    var data by this.itemStack.component(EAEPDataComponents.Priority())
}
